Compiling errors for extension footprint_extruder

Thanks for your fantastic work!

I was trying to set up the environment for the demo but ran into compiling errors when installing `extensions/footprint_extruder`

```cpp
(city) username@server:~/source/GaussianCity/extensions/footprint_extruder$ pip install .
Processing /home/username/source/GaussianCity/extensions/footprint_extruder
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: footprint_extruder
  Building wheel for footprint_extruder (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [9 lines of output]
      footprint_extruder.cpp: In function 'int initNumpy()':
      footprint_extruder.cpp:29:3: warning: converting to non-pointer type 'int' from NULL [-Wconversion-null]
         29 |   import_array();
            |   ^~~~~~~~~~~~
      footprint_extruder.cpp: In function 'bool isNeighboringValueSame(const short int*, short int, short int, short int, short int)':
      footprint_extruder.cpp:103:24: error: variable 'std::array<short int, 8> nbr_values' has initializer but incomplete type
        103 |   std::array<short, 8> nbr_values{
            |                        ^~~~~~~~~~
      error: command '/home/username/miniconda3/envs/city/bin/x86_64-conda-linux-gnu-c++' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for footprint_extruder
  Running setup.py clean for footprint extruder
Failed to build footprint_extruder
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (footprint_extruder)
```

I was compiling the code with g++ 13.3.0 installed via conda on a Ubuntu 24.04.1 server,

It seems that `std::array` is used without including the header `<array>`. The compiling error was gone when I added `#include <array>`  to the beginning of `footprint_extruder.cpp`.

I also notice that `std::numeric_limits` is used without including `<limits>`. And for the extension `grid_encoder`,  `<stdint.h>` is included instead of the cpp counterpart `<cstdint>`. Although depending on compiler versions, errors may occur or not, I hope there could be a fix to follow the best practices of C++. I can make a pull request for these fixes.

--- a/extensions/footprint_extruder/footprint_extruder.cpp
+++ b/extensions/footprint_extruder/footprint_extruder.cpp
@@ -13,8 +13,10 @@
 #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
 #define N_DIM 5
 
+#include <array>
 #include <cassert>
 #include <cmath>
+#include <limits>
 #include <map>
 #include <string>
 #include <vector>

--- a/extensions/grid_encoder/bindings.cpp
+++ b/extensions/grid_encoder/bindings.cpp
@@ -8,7 +8,7 @@
  * @Ref: https://github.com/ashawkey/torch-ngp
  */
 
-#include <stdint.h>
+#include <cstdint>
 #include <torch/extension.h>
 #include <torch/torch.h>
 

--- a/extensions/grid_encoder/grid_encoder_ext.cu
+++ b/extensions/grid_encoder/grid_encoder_ext.cu
@@ -17,9 +17,10 @@
 
 #include <algorithm>
 #include <stdexcept>
+#include <type_traits>
 
 #include <cstdio>
-#include <stdint.h>
+#include <cstdint>
 
 #define CHECK_CUDA(x)                                                          \
   TORCH_CHECK(x.device().is_cuda(), #x " must be a CUDA tensor")

